dist_tables/ddl.rst

```diff
@@ -148,3 +148,135 @@ You can use the standard PostgreSQL DROP TABLE command to remove your distribute
+
+This example, excerpted from a :ref:`typical_mt_schema`, shows how to create primary and foreign keys on distributed tables.
+
+.. code-block:: postgresql
+
+  --
+  -- Adding a primary key
+  -- --------------------
+
+  -- Ultimately we'll distribute these tables on the account id, so the
+  -- ads and clicks tables use compound keys to include it.
+
+  ALTER TABLE accounts ADD PRIMARY KEY (id);
+  ALTER TABLE ads ADD PRIMARY KEY (account_id, id);
+  ALTER TABLE clicks ADD PRIMARY KEY (account_id, id);
+
+  -- Next distribute the tables
+  -- (primary keys must be created prior to distribution)
+
+  SELECT create_distributed_table('accounts',  'id');
+  SELECT create_distributed_table('ads',       'account_id');
+  SELECT create_distributed_table('clicks',    'account_id');
+
+  --
+  -- Adding foreign keys
+  -- -------------------
+
+  -- Note that this can happen before or after distribution, as long as
+  -- there exists a uniqueness constraint on the target column(s) which
+  -- can only be enforced before distribution.
+
+  ALTER TABLE ads ADD CONSTRAINT ads_account_fk
+    FOREIGN KEY (account_id) REFERENCES accounts (id);
+  ALTER TABLE clicks ADD CONSTRAINT clicks_account_fk
+    FOREIGN KEY (account_id) REFERENCES accounts (id);
+
+Uniqueness constraints, like primary keys, must be added prior to table distribution.
+
+.. code-block:: postgresql
+
+  -- Suppose we want every ad to use a unique image. Notice we can
+  -- enforce it only per account when we distribute by account id.
+
+  ALTER TABLE ads ADD CONSTRAINT ads_unique_image
+    UNIQUE (account_id, image_url);
+
+Not-null constraints can always be applied because they require no lookups between workers.
+
+.. code-block:: postgresql
+
+  ALTER TABLE ads ALTER COLUMN image_url SET NOT NULL;
+
+Adding/Removing Indices
+~~~~~~~~~~~~~~~~~~~~~~~
+
+Citus supports adding and removing `indices <https://www.postgresql.org/docs/current/static/sql-createindex.html>`_, but not with the :code:`CONCURRENTLY` option.
+
+.. code-block:: postgresql
+
+  -- Adding an index (does not support CONCURRENTLY)
+
+  CREATE INDEX clicked_at_idx ON clicks USING BRIN (clicked_at);
+
+  -- Removing an index
+
+  DROP INDEX clicked_at_idx;
+
+Manual Modification
+~~~~~~~~~~~~~~~~~~~
+
+If you attempt to alter a table and get an error such as "ERROR:  0A000: cannot create constraint," it means Citus does not yet support auto-propagation for the operation. In this case you can propagate the changes manually using this general four-step outline:
```

Comment:
Is this the actual error we get? It seems specific to constraints, and I'd expect we should have a more proper Citus error around not being able to propagate the command. Could we remove this, and replace with something like "Currently other DDL commands are not auto-propagated, however there are different workarounds to manually propagate these to worker nodes" (or something like that).